lua/cfc_ulx_commands/curse/effects/random_sounds.lua

```diff
@@ -0,0 +1,142 @@
+local EFFECT_NAME = "RandomSounds"
+local POOL_SIZE = 100 -- Smaller pool sizes help reduce lag from loading new sounds on the fly (most notable with footsteps).
+local POOL_BATCH_SIZE = 10 -- Builds up the pool in batches over time, since it requires sounds to be loaded to see if they're good.
+local POOL_BATCH_INTERVAL = 1
+local SOUND_DURATION_MAX = 5
+
+
+-- Create global table
+local EFFECT_NAME_LOWER = string.lower( EFFECT_NAME )
+CFCUlxCurse.EffectGlobals[EFFECT_NAME_LOWER] = CFCUlxCurse.EffectGlobals[EFFECT_NAME_LOWER] or {}
+local globals = CFCUlxCurse.EffectGlobals[EFFECT_NAME_LOWER]
+
+local allSounds = nil
+local allSoundsLength = nil
+local poolSounds = {}
+local poolSoundsLength = nil
+local poolSoundsTargetLength = nil
+local entityMeta = FindMetaTable( "Entity" )
+
+local mathRandom = math.random
+local stringFind = string.find
+local stringSub = string.sub
+local soundDuration = SoundDuration
+
+
+local function getSound()
```

Comment:
It's kind of confusing in here but I think this is clearer?
```suggestion
local function getPoolSound()
```